libgadget/drift.c

```diff
@@ -110,6 +110,10 @@ static void real_drift_particle(int i, inttime_t ti1, const double ddrift, const
         if(fac > 1.25)
             fac = 1.25;
         P[i].Hsml *= fac;
+        /* Cap the Hsml: if DivVel is large for a particle with a long timestep
+         * (most likely a wind particle) Hsml can very rarely run away*/
+        if(P[i].Hsml > All.BoxSize /2.)
```

Comment:
Isn't half of box a bit too large? I remember we used something like 30 Kpc/h in MB-II for the hsml cap at some point.

With such large hsml the particle will interact with every particle in the entire simulation -- so it must be exported to all ranks.

Reply:
Yes, I think you are right! I think the right thing is to scale with the MeanInterparticleSpacing. Perhaps DMIN(MeanInterparticleSpacing * 100, All.BoxSize)?

Reply:
I am a bit short circuited here. Looks like the factor (50) translate to the density / mean density ratio -- so if you know the target lowest baryon density you want to model, then you can work out the number.

Reply:
That's right - I was thinking we could bound it using a lognormal density distribution. But I am unsure that this is the right thing to do. All.BoxSize/2 has the advantage that it is the largest physically sensible value for any particle distribution, even pathological ones where, eg, one particle is in the middle and all the others are at the box edges.

All the other choices could potentially give the wrong answer in some subtle way in a future paper studying deep voids, or easily hide real bugs. So I actually think that BoxSize/2 is the right choice.